lnrpc/rpc.proto

```diff
@@ -1114,31 +1114,31 @@ message HTLC {
 }
 
 enum CommitmentType {
+    /*
+    Returned when the commitment type isn't known or unavailable.
+    */
+    UNKNOWN_COMMITMENT_TYPE = 0;
```

Comment:
Agreed, which is why I went with this approach. Also, before this change, the `CommitmentType` message is only ever included in responses and not requests.

Reply:
It's a breaking change from the PoV of a client that has an older set of protos, and uses the existing enums/constants to make an assertion against these new values. However it was only used in `listchannels` and `pendingchannels`. 

@wpaulino so the rationale w.r.t making the zero type "unknown" is that we can eventually enforce that users always set a type explicitly? How will a "default" value be added? We could instead go with a "this value is the default" approach similar to what we went w/ in Pool w.r.t selecting node tier.

Reply:
> @wpaulino so the rationale w.r.t making the zero type "unknown" is that we can eventually enforce that users always set a type explicitly? How will a "default" value be added? We could instead go with a "this value is the default" approach similar to what we went w/ in Pool w.r.t selecting node tier.

We want to have an "unknown" variant as the zero value so that we can easily detect when a user hasn't provided a channel type explicitly. This also allows us to set whatever default we deem reasonable from the RPC server without having to make proto changes.

We can achieve this in two ways:

1. Rework the existing `CommitmentType` proto to have this unknown variant, and be comfortable with breaking it.
2. Add a new `CommitmentType` proto alternative and deprecate the old one.

Reply:
Gotcha, I think we're using unknown/default in the same manner, need to circle back on the diff, but I think the current approach will work, assuming we map "unknown" to w/e out default is at the time.